integration_tests/e2e/make-a-referral.cy.ts

```diff
@@ -1,20 +1,243 @@
+  cy.findByRole('textbox', { name: /date of incident/i }).type('06/07/2024')
+  cy.findByRole('textbox', { name: /hour/i }).type('12')
+  cy.findByRole('textbox', { name: /minute/i }).type('24')
+  cy.findByRole('combobox', { name: /where did the incident occur\?/i }).select('LocationB')
+  cy.findByRole('combobox', { name: /what was the incident type\?/i }).select('TypeA')
+  checkAxeAccessibility()
+  cy.findByRole('button', { name: /continue/i }).click()
+
+  cy.url().should('include', '/involvement')
+  cy.findByRole('heading', { name: /incident involvement/i }).should('be.visible')
+  cy.findByRole('radio', { name: /factor1/i }).click()
+  cy.findByRole('radio', { name: /yes/i }).click()
+  cy.findByRole('textbox', { name: /names of staff assaulted/i }).type('TestStaff Member')
+  checkAxeAccessibility()
+  cy.findByRole('button', { name: /continue/i }).click()
+
+  cy.url().should('include', '/description')
+  cy.findByRole('heading', { name: /describe the incident and concerns/i }).should('be.visible')
+  // Cypress does not check the visibility of elements inside <details> elements properly, so use browser builtins instead
+  cy.findByText(/the description could include/i).then(el => {
+    expect(el.get(0).checkVisibility()).to.eq(false)
+  })
+  cy.findByText(/what type of information to include/i).click()
+  cy.findByText(/the description could include/i).then(el => {
+    expect(el.get(0).checkVisibility()).to.eq(true)
+  })
+  cy.findByRole('textbox', { name: /describe the incident and concerns/i }).type('incident concerns foobar123')
+  checkAxeAccessibility()
+  cy.findByRole('button', { name: /continue/i }).click()
+
+  cy.url().should('include', '/reasons')
+  cy.findByRole('heading', { name: /what reasons have been given for the incident\?/i }).should('be.visible')
+  cy.findByText(/include any reasons given/i).then(el => {
+    expect(el.get(0).checkVisibility()).to.eq(false)
+  })
+  cy.findByText(/what type of information to include/i).click()
+  cy.findByText(/include any reasons given/i).then(el => {
+    expect(el.get(0).checkVisibility()).to.eq(true)
+  })
+  cy.findByRole('textbox', { name: /what reasons have been given for the incident/i }).type(
+    'incident reasons foobar123',
+  )
+  checkAxeAccessibility()
+  cy.findByRole('button', { name: /continue/i }).click()
+
+  cy.url().should('include', '/contributory-factors')
+  cy.findByRole('heading', { name: /what are the contributory factors\?/i }).should('be.visible')
+  cy.findByRole('checkbox', { name: /factor1/i }).click()
+  cy.findByRole('checkbox', { name: /factor3/i }).click()
+  checkAxeAccessibility()
+  cy.findByRole('button', { name: /continue/i }).click()
+
+  cy.url().should('include', '/contributory-factors-comments')
+  cy.findAllByRole('link', { name: /add comment/i }).should('have.length', 2)
+  cy.findAllByRole('link', { name: /add comment/i })
+    .first()
```

Comment:
i think you can `findByRole('link', { name: /add comment on factor1 factor/i })` here

Reply:
Ooh thanks I'll try that